raycicmd/command_step.go

```diff
@@ -0,0 +1,145 @@
+package raycicmd
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+)
+
+type commandConverter struct {
+	config *config
+	info   *buildInfo
+
+	envMap map[string]string
+}
+
+func newCommandConverter(
+	config *config, info *buildInfo, envMap map[string]string,
+) *commandConverter {
+	return &commandConverter{
+		config: config,
+		info:   info,
+		envMap: envMap,
+	}
+}
+
+func (c *commandConverter) mapAgent(instanceType string) (string, error) {
+	if instanceType == "" {
+		instanceType = "default"
+	}
+	if q, ok := c.config.RunnerQueues[instanceType]; ok {
+		return q, nil
+	}
+	return "", fmt.Errorf("unknown instance type %q", instanceType)
+}
+
+func (c *commandConverter) jobEnvImage(name string) string {
+	if name == "" {
+		name = "forge"
+	}
+
+	return fmt.Sprintf("%s:%s-%s", c.config.CIWorkRepo, c.info.buildID, name)
+}
+
+const dockerPlugin = "docker#v5.8.0"
+const macosSandboxPlugin = "ray-project/macos-sandbox#v1.0.7"
+
+func (c *commandConverter) match(step map[string]any) bool { return true }
```

Comment:
why is this always true?

Reply:
it is the default converter that converts anything when it does not mach any other converter.

I can make this stricter, but the match function is actually not executed, I fear that making changes to the behavior will make things break (e.g. some weird steps that take weird forms).